src/main/java/roboy/dialog/states/ordinaryStates/IntroductionState.java

```diff
@@ -0,0 +1,156 @@
+        return Output.say("I'm Roboy. What's your name?");
+    }
+
+    @Override
+    public Output react(Interpretation input) {
+
+        // expecting something like "I'm NAME"
+
+        // 1. get name
+        String name = getNameFromInput(input);
+
+        if (name == null) {
+            // input couldn't be parsed properly
+            // TODO: do something intelligent if the parser fails
+            nextState = this;
+            logger.warn("IntroductionState couldn't get name! Staying in the same state.");
+            return Output.say("Sorry, my parser is out of service.");
+            // alternatively: Output.useFallback() or Output.sayNothing()
+        }
+
+
+        // 2. get interlocutor object from context
+        // this also should query memory and do other magic
+        Interlocutor person = Context.getInstance().ACTIVE_INTERLOCUTOR.getValue();
+        person.addName(name);
+
+
+        // 3. update interlocutor in context
+        updateInterlocutorInContext(person);
+
+        // 4. check if person is known/familiar
+        if (person.FAMILIAR) {
+
+            // 4a. person known/familiar
+            String retrievedResult = "";
+            RandomList<MemoryNodeModel> nodes = retrieveNodesFromMemoryByIds(person.getRelationships(Neo4jRelationships.FRIEND_OF));
+            if (!nodes.isEmpty()) {
+                retrievedResult = " You are friends with " + nodes.getRandomElement().getProperties().get("name").toString();
+            }
+
+            RelationshipAvailability availability = person.checkRelationshipAvailability(predicates);
+            if (availability == SOME_AVAILABLE) {
+                nextState = (Math.random() < 0.3) ? getTransition(UPDATE_KNOWN_PERSON) : getTransition(LEARN_ABOUT_PERSON);
+            } else if (availability == NONE_AVAILABLE) {
+                nextState = getTransition(LEARN_ABOUT_PERSON);
+            } else {
+                nextState = getTransition(UPDATE_KNOWN_PERSON);
+            }
+
+            // TODO: get some friends or hobbies of the person to make answer more interesting
+            return Output.say("Hey, I know you, " + person.getName() + retrievedResult);
+
+        } else {
+            // 4b. person is not known
+            nextState = getTransition(LEARN_ABOUT_PERSON);
+
+            // TODO: what would you say to a new person?
+            return Output.say(String.format("Nice to meet you, %s!", name)).setSegue(new Segue(Segue.SegueType.DISTRACT, 0.6));
+        }
+    }
+
+    @Override
+    public State getNextState() {
+        return nextState;
+    }
+
+    private String getNameFromInput(Interpretation input) {
+        if (input.getSentenceType().compareTo(Linguistics.SENTENCE_TYPE.STATEMENT) == 0) {
+            String[] tokens = (String[]) input.getFeatures().get(Linguistics.TOKENS);
+            if (tokens.length == 1) {
+                return tokens[0].replace("[", "").replace("]","").toLowerCase();
```

Comment:
I think these replacements should be done by the parser (or at least in the ``SemanticParserAnalyzer``). There is no reason why tokens provided to the dialog system should contain ``[]``.

Reply:
@emlozin please take a look